### PR TITLE
PLAT-846: validate methods for VCachedMemoryRequest/Response

### PR DESCRIPTION
**Pull request by aalekseev-ru**
Add VCachedMemoryRequest/VCachedMemoryResponse Validate functions

### CODE REVIEW

File: ledger-core/rms/proto_vnvn_validation_vcachedmemoryrequest.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import "github.com/insolar/assured-ledger/ledger-core/vanilla/throw"
+
+func (m *VCachedMemoryRequest) Validate(currentPulse PulseNumber) error {
+	object := m.Object.GetValue()
+	_, err := validSelfScopedGlobalWithPulseBeforeOrEq(object, currentPulse, "Object")
+	if err != nil {
+		return err
+	}
+
+	if m.Object.IsEmpty() {
```
Comment (ElSacramento) [inline ledger-core/rms/proto_vnvn_validation_vcachedmemoryrequest.go:17]: I suppose, `validSelfScopedGlobalWithPulseBeforeOrEq` will fail if object is empty.

File: ledger-core/rms/proto_vnvn_validation_vcachedmemoryresponse.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+func (m *VCachedMemoryResponse) Validate(currentPulse PulseNumber) error {
+	object := m.Object.GetValue()
+	_, err := validSelfScopedGlobalWithPulseBeforeOrEq(object, currentPulse, "Object")
+	if err != nil {
+		return err
+	}
+
+	switch m.CallStatus {
+	case CachedMemoryStateFound:
+		if m.Memory.IsEmpty() {
```
Comment (ElSacramento) [inline ledger-core/rms/proto_vnvn_validation_vcachedmemoryresponse.go:21]: `Class` isn't empty, too

Comment (aalekseev-ru) [reply ledger-core/rms/proto_vnvn_validation_vcachedmemoryresponse.go:21]: m.Class doens't exist at master

Comment (ElSacramento) [reply ledger-core/rms/proto_vnvn_validation_vcachedmemoryresponse.go:21]: and object is connected to the class, but I don't know how to check it

Comment (ElSacramento) [reply ledger-core/rms/proto_vnvn_validation_vcachedmemoryresponse.go:21]: =( add todo, please

File: ledger-core/rms/proto_vnvn_validation_vcachedmemoryrequest.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import "github.com/insolar/assured-ledger/ledger-core/vanilla/throw"
+
+func (m *VCachedMemoryRequest) Validate(currentPulse PulseNumber) error {
+	object := m.Object.GetValue()
+	_, err := validSelfScopedGlobalWithPulseBeforeOrEq(object, currentPulse, "Object")
+	if err != nil {
+		return err
+	}
+
+	if m.StateID.IsEmpty() {
```
Comment (ElSacramento) [inline ledger-core/rms/proto_vnvn_validation_vcachedmemoryrequest.go:17]: todo about "stateID and object are from one chain"

Comment (aalekseev-ru) [reply ledger-core/rms/proto_vnvn_validation_vcachedmemoryrequest.go:18]: done

File: ledger-core/rms/proto_vnvn_validation_vcachedmemoryresponse.go (hunk 2 above)
Comment (aalekseev-ru) [reply ledger-core/rms/proto_vnvn_validation_vcachedmemoryresponse.go:21]: done

**Review by ElSacramento — DISMISSED** [review]

**Review by ruz — APPROVED** [review]

**Review by ElSacramento — APPROVED** [review]

**Pull request merged**